js/packages/common/src/contexts/connection.tsx

```diff
@@ -656,71 +656,68 @@ async function awaitTransactionSignatureConfirmation(
     err: null,
   };
   let subId = 0;
-  status = await new Promise(async (resolve, reject) => {
+  status = await (async (): Promise<SignatureStatus | null | void> => {
```

Comment:
Only `connection.onSignature()` requires the `new Promise` treatment to make it compatible with async/await, but we should be able to implement the rest of this code as an async lambda. I _think_ this makes it easier to follow, but let me know if I missed or messed anything up.